### ISSUE TITLE
testrace error: closing closed channel in multiraft.(*state).handleWriteResponse

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#307]
relevant bit, from local `make testrace` output (on my feature branch, but I think it's independent)
```
panic: close of closed channel
goroutine 229 [running]:
github.com/cockroachdb/cockroach/multiraft.(*state).handleWriteResponse(0xc2083a3b90, 0xc208044ad8, 0xc2085128d0)
        /home/tobias/Code/Go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:706 +0x878
github.com/cockroachdb/cockroach/multiraft.(*state).start(0xc2083a3b90)
        /home/tobias/Code/Go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:454 +0x1c35
created by github.com/cockroachdb/cockroach/multiraft.(*MultiRaft).Start
        /home/tobias/Code/Go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:147 +0x60
```

[full text here](https://gist.github.com/tschottdorf/b8db4fd932a445a2b670)

**Comment by bdarnell** [comment]
Hmm, I thought I fixed that in 25342e8b. Can you confirm that you have that commit in your branch and you're still seeing this error?

**Comment by tschottdorf** [comment]
It's the state of coalesced_heartbeats_rb as of writing. Will check when at
home but almost certainly.
